.github/workflows/run-tests.yml

```diff
@@ -212,7 +259,7 @@ jobs:
           docker compose -f ${{ env.DOCKER_COMPOSE_FILE }} up -d ${{ join(matrix.containers, ' ') }}
 
       - name: Run functional test
-        timeout-minutes: 30 # make sure this is larger than the test timeout in the Makefile
+        timeout-minutes: ${{ fromJSON(env.FUNCTIONAL_TEST_TIMEOUT) }} # make sure this is larger than the test timeout in the Makefile
```

Comment:
Doesn't this work?
```suggestion
        timeout-minutes: ${{ needs.set-up-functional-test.outputs.functional_test_timeout }} # make sure this is larger than the test timeout in the Makefile
```

Reply:
that does not work, because github requires that the field for `timeout-minutes` be a number, not a string. `fromJSON()` converts it to a number. 

The code is now changed so that the `set-up-functional-test` step produces two outputs, which are based on the input timeout, but slightly different, since `go test` needs a time string, and github action needs a number, and the github actions timeout should be higher than the go timeout:
```
eg. input_timeout=10

github_timeout: "15"
test_timeout: "10m"
```
and then I pass the github timeout directly to the `timeout-minutes` field, and the test_timeout to the env var used by the makefile

```
TEST_TIMEOUT: ${{ needs.set-up-functional-test.outputs.test_timeout }}
...
timeout-minutes: ${{ fromJSON(needs.set-up-functional-test.outputs.github_timeout) }} # make sure this is larger than the test timeout in the Makefile
```